services/blockstorage/test/block_sync_test.go

```diff
@@ -196,8 +208,43 @@ func TestSyncCompletePetitionerSyncFlow(t *testing.T) {
 		harness.expectCommitStateDiffTimes(4)
 		harness.expectValidateWithConsensusAlgosTimes(4)
 
+		// fake the response
 		harness.blockStorage.HandleBlockSyncResponse(blockSyncResponse)
 
+		// verify that we committed the blocks
 		harness.verifyMocks(t, 4)
 	})
 }
+
+func TestSyncNeverStartsWhenBlocksAreCommitted(t *testing.T) {
+	test.WithContext(func(ctx context.Context) {
+		// let the sync time to start
+		time.Sleep(1 * time.Millisecond)
```

Comment:
`*ugh*`

Reply:
tell me about it..

Reply:
can you think of a way to make this deterministic?

Reply:
if i had a way i would have done it by now. 
there is one thing in the block-sync design that 'troubled' me, is that when i shut it down, in order to know its actually shut down i am holding a boolean, it feels like something is missing in that area, like a letch saying 'good to go', thats my only idea around that.